src/Box.hpp

```diff
@@ -31,13 +31,26 @@ class Box
     // Draw the box
     virtual void draw(sf::RenderWindow& renderWindow, std::unordered_set<std::string>& choiceHistory) = 0;
 
+    // Request a redraw of the box. Subclasses should call this when the graphics change
+    void requestDraw();
+
+    // Indicate to the box that drawing has completed. Subclasses should not
+    // need to call this.
```

Comment:
In which case this should be private?

Reply:
The way I designed it, this was intended to be called by other code that was not a subclass of this one (like GameSlice, or in the case of GameSlice, the main event loop in StoryTime.cpp). If you would rather have each class call this at the end of drawing (as you suggest below), then we can change the comment.